Extensions/Lighting/lightruntimeobject-pixi-renderer.js

```diff
@@ -0,0 +1,254 @@
+    this._light = new PIXI.Mesh(this._geometry, this._shader);
+  }
+  this._light.blendMode = PIXI.BLEND_MODES.ADD;
+
+  if (this.DEBUG) {
+    if (this._graphics === undefined) {
+      this._graphics = new PIXI.Graphics();
+      this._graphics
+        .lineStyle(1, 0xff0000, 1)
+        .moveTo(this._object.x, this._object.y)
+        .lineTo(this._object.x - this._radius, this._object.y + this._radius)
+        .moveTo(this._object.x, this._object.y)
+        .lineTo(this._object.x + this._radius, this._object.y + this._radius)
+        .moveTo(this._object.x, this._object.y)
+        .lineTo(this._object.x + this._radius, this._object.y - this._radius)
+        .moveTo(this._object.x, this._object.y)
+        .lineTo(this._object.x - this._radius, this._object.y - this._radius);
+    }
+    if (this._debugLight === undefined) {
+      this._debugLight = new PIXI.Container();
+      this._debugLight.addChild(this._light);
+      this._debugLight.addChild(this._graphics);
+    }
+  }
+};
+
+gdjs.LightRuntimeObjectRenderer = gdjs.LightRuntimeObjectPixiRenderer; //Register the class to let the engine use it.
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.getRendererObject = function () {
+  // Mandatory, return the internal PIXI object used for your object:
+  if (this.DEBUG) {
+    return this._debugLight;
+  }
+  return this._light;
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.ensureUpToDate = function () {
+  if (this.DEBUG) this.updateGraphics();
+  this.updateVertexBuffer();
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.updateGraphics = function () {
+  var vertices = [this._object.x, this._object.y];
+  var raycast = this.raycastTest().reduce(function (acc, val) {
+    return acc.concat(val);
+  });
+  vertices.push.apply(vertices, raycast);
+
+  this._graphics.clear();
+  for (var i = 0; i < vertices.length; i += 2) {
+    var lineColor = i % 4 === 0 ? 0xff0000 : 0x00ff00;
+    this._graphics
+      .lineStyle(1, lineColor, 1)
+      .moveTo(vertices[0], vertices[1])
+      .lineTo(vertices[i], vertices[i + 1]);
+    if (i !== vertices.length - 2) {
+      this._graphics.lineTo(vertices[i + 2], vertices[i + 3]);
+    } else {
+      this._graphics.lineTo(vertices[2], vertices[3]);
+    }
+  }
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.updateVertexBuffer = function () {
+  this._light.shader.uniforms.center = new Float32Array([
+    this._object.x,
+    this._object.y,
+  ]);
+
+  var raycast = this.raycastTest();
+
+  var vertexBuffer = [this._object.x, this._object.y];
+
+  for (var i = 2; i < 2 * raycast.length + 2; i += 2) {
+    vertexBuffer[i] = raycast[i / 2 - 1][0];
+    vertexBuffer[i + 1] = raycast[i / 2 - 1][1];
+  }
+
+  var indexBuffer = [];
+
+  for (var i = 0; i < 3 * raycast.length; i += 3) {
+    indexBuffer[i] = 0;
+    indexBuffer[i + 1] = i / 3 + 1;
+    if (i / 3 + 1 !== raycast.length) indexBuffer[i + 2] = i / 3 + 2;
+    else indexBuffer[i + 2] = 1;
+  }
```

Comment:
It's hard to follow up what's going on here without knowing about the algorithm. Can you add comments for each of the two loop describing what you're constructing?

Reply:
I'll add comments before the loops :) The basic idea is this: 

1. Vertex buffer contains all the points we get after raycasting. Let's take the example of simplest case, when we only have the 4 vertices of quad. 

0th vertex is the center, we then have the other 4 points. This means we store 5 vertices. Therefore the size of vertex buffer = `2 * 4 + 2 = 10`. (4 = number of vertices we get from raycasting, we have factors of 2 here because each vertex has x and y, `+ 2` comes from having center as 0th vertex).

2. Index buffer follows a simple pattern ;) We need to render triangles, so the pattern goes like this: 0, 1, 2, 0, 2, 3, 0, 3, 4, 0, 4, 1 (where 0 represents center, and this pattern forms triangles with center and two adjacent vertices) Also the last triangle should end up at the first vertex to make a close polygon.

Reply:
Sounds good, a similar short but helpful explanation in the comment can goes a long way in allowing a reader to understand what's going on :) 
Thinking of indexes, we should give another try at the "Triangle Fan" draw mode, in theory this can reduce by two (more or less?) the number of indexes/vertex because we don't need to repeat the center (nor the previous points).

Reply:
Yes, I'm going to try triangle fans once more. If it works out there will be a huge benefit to memory (even on the GPU side). We wouldn't need the index buffer at all, and there will be `n + 2` vertices in vertex buffer, n being the number of vertices calculated by the raycasting algo.

Reply:
Would be awesome if we can get this to work 👍

Reply:
I tried this again, didn't work :( It renders with horrible artifacts 😓 (and sometimes doesn't show up on the screen at all).